dev/com.ibm.ws.security.wim.adapter.ldap/src/com/ibm/ws/security/wim/adapter/ldap/context/ContextManager.java

```diff
@@ -236,6 +250,22 @@
     /** Write to secondary server configuration. */
     private boolean iWriteToSecondary = false;
 
+    /** The Ldap repo ID for use in logging **/
+    private String reposId = null;
+
+    /** Sets the bind authentication mechanism for administrative binds to the Ldap server. None, simple, GSSAPI (Kerbers). */
+    private String bindAuthMechanism = null;
+
+    /** GSSAPI/Kerberos configuration. krb5Principal is required, krb5TicketCache is optional. Keytab/config can be configured in the <kerberos> element */
+    private String krb5Principal = null;
+    private Path krb5TicketCache = null;
+
+    /** The KerberosService for use when bindAuthMechanism is GSSAPI (Kerberos), also loads the keytab/config if configured in the <kerberos> element */
+    private KerberosService kerberosService = null;
+
+    /** Read/Write lock to prevent multiple processes from using and updating the KerberosService and/or context pool at the same time. **/
+    private final ReadWriteLock kerberServiceModifyLock = new ReentrantReadWriteLock();
```

Comment:
Currently you only use the write lock, which means that none of the code blocked is reentrant. Did we intend to use the readLock in handleKerberos? I am not sure if we can / should have multiple threads in there at once.

If we aren't using the readLock, then we may want to choose a different locking mechanism.

Reply:
Doh, yes. Swapping handleKerberos to use a read lock.